extensions/kinesis/src/main/java/com/hazelcast/jet/kinesis/KinesisSources.java

```diff
@@ -286,16 +290,40 @@ public Builder withInitialShardIteratorRule(@Nonnull String shardIdRegExp,
          */
         @Nonnull
         public StreamSource<Map.Entry<String, byte[]>> build() {
+            FunctionEx<? super Record, byte[]> projectionFn = Builder::toArray;
```

Comment:
I am not an expert on Kinesis. Is the key here important? Could the default projection function be Record -> Entry<String, byte[]>, allowing the user to provide function from Record -> CustomType, resulting in a source emitting items without the Entry wrapper? That's what we how the types look with most of the connectors with custom `projectionFn`

Reply:
After giving it a thought, you are right! Projection Function can produce target type, without wrapping, as not everyone will be interested in the Kinesis partitionKey.
I'll also add that projection function receives Record and Shard info. Otherwise the ``withInitialShardIteratorRule`` has no sense - how you would like to start from last sequence number if you don't have sequence number per shard? It will make a lot of sense for many people I think. I'll change it today after work :)